### ISSUE TITLE
Support for codex32/BIP93 seed import

### ISSUE THREAD

**Issue by apoelstra** [bitcoindevkit/bdk#1027]
Hi all,

I would like to support importing seeds in the BIP93 format, which consists of either
* A bech32-encoded BIP32 seed, of length 128-512 bits, with a 13-character checksum (15 characters for 300 bits or more)
* A collection of bech32-encoded BIP32 seed *shares*, which can be reconstructed into a seed using SSSS.

I am happy to do 100% of the implementation and integration work. We will require a new dependency, [rust-codex32](
https://github.com/apoelstra/rust-codex32), though I am working on the existing dependency rust-bech32 to make it generic over checksum types, which may make rust-codex32 small enough that we can just inline the essential funcitonality.

See https://secretcodex32.com/index.html for more information.

Would people be interested/accepting of this?

**Comment by notmandatory** [comment]
This looks like a valuable addition to me. 

If it adds a new dependency I think it'd be best to add a new optional feature like what we did for bip39, but if it becomes part of `rust-bech32` it shouldn't need a feature flag. Will this only require a little glue code like `impl<Ctx: ScriptContext> DerivableKey<Ctx> for Codex32String` ?

**Comment by apoelstra** [comment]
@notmandatory yeah, not much more than that. Though it won't be `DerivableKey` but probably something else because codex32 seeds are BIP32 seeds rather than keys. (The same applies to bip39 seed words; I'll just copy whatever API exists for that.)

Sure, I'm happy to have a feature gate as long as we need an extra dep.
